File: 2024-02-05-problems-when-make-test-dobule-in-kotlin/action-in-blog/java-in-action/src/main/java/blog/in/action/service/TodoService.java
package blog.in.action.service;

import org.springframework.stereotype.Service;

import java.time.LocalDate;
import java.util.List;

@Service
public class TodoService {

    public List<String> getTodos() {
        return List.of("Hello", "World");
    }

    public String getTodo(long id) {
        return "Hello";
    }

    public List<String> getTodosInToday(LocalDate localDate) {
        return List.of("Hello", "World", localDate.toString());
    }
}

File: 2024-02-05-problems-when-make-test-dobule-in-kotlin/action-in-blog/kotlin-in-action/src/main/kotlin/blog/in/action/service/TodoService.kt
package blog.`in`.action.service

import org.springframework.stereotype.Service
import java.time.LocalDate

@Service
class TodoService {

    fun getTodos(): List<String> {
        return listOf("Hello", "World")
    }

    fun getTodo(id: Long): String {
        return "Hello"
    }

    fun getTodosInToday(localDate: LocalDate?): List<String> {
        return listOf("Hello", "World", localDate.toString())
    }

    fun getFilteredTodosInToday(ids: List<Long>, localDate: LocalDate?): List<String> {
        return listOf("Hello", "World", ids.size.toString(), localDate.toString())
    }
}